plugins/new-relic/retention.yaml

```diff
@@ -1013,3 +1013,217 @@ presetScripts:
+          df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
+
+          # Set client and server based on trace_role.
+          df.source_pod = px.select(df.is_server_tracing, df.ra_name, df.pod)
+          df.destination_pod = px.select(df.is_server_tracing, df.pod, df.ra_name)
+
+          df = df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])
+          df.source_service = px.pod_name_to_service_name(df.source_pod)
+          df.destination_service = px.pod_name_to_service_name(df.destination_pod)
+          df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
+          df.source_namespace = px.pod_name_to_namespace(df.source_pod)
+
+          df.source_service = remove_ns_prefix(df.source_service)
+          df.destination_service = remove_ns_prefix(df.destination_service)
+          df.source_pod = remove_ns_prefix(df.source_pod)
+          df.destination_pod = remove_ns_prefix(df.destination_pod)
+
+          # If the destination service is missing then try the nslookup
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              px.nslookup(df.remote_addr),
+          )
+          # If the destination service is still missing then set the remote_addr
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              df.remote_addr,
+          )
+          return df
+
+      df = px.DataFrame('cql_events', start_time=px.plugin.start_time, end_time=px.plugin.end_time)
+      df = add_source_dest_columns(df)
+
+      df.start_time = df.time_ - df.latency
+      df.latency = df.latency / (1000 * 1000)
+      df.req_bytes = px.length(df.req_body)
+      df.resp_bytes = px.length(df.resp_body)
+      df = df.groupby(['source_pod', 'source_service', 'source_namespace', 'destination_pod', 'destination_service',
+                        'destination_namespace', 'req_op']).agg(
+          latency_count=('latency', px.count),
+          latency_min=('latency', px.min),
+          latency_max=('latency', px.max),
+          latency_sum=('latency', px.sum),
+          time_=('time_', px.max),
+          req_bytes=('req_bytes', px.sum),
+          resp_bytes=('resp_bytes', px.sum),
+      )
+
+      df.pixie = 'pixie'
+      df.cluster_id = px.vizier_id()
+      df.cluster_name = px.vizier_name()
+      df.db_system = 'cassandra'
+
+      px.export(
+        df, px.otel.Data(
+          resource={
+            'service.name': df.source_service,
+            'service.instance.id': df.source_pod,
+            'k8s.namespace.name': df.source_namespace,
+            'cassandra.service.name': df.destination_service,
+            'cassandra.pod.name': df.destination_pod,
+            'cassandra.namespace.name': df.destination_namespace,
+            'cassandra.req_op': df.req_op,
```

Comment:
Note: we only include the `req_op` and not the `req_body` because the later can be large.